pkg/skaffold/kubernetes/status/status_check.go

```diff
@@ -71,40 +75,65 @@ type Config interface {
 	GetNamespaces() []string
 	StatusCheckDeadlineSeconds() int
 	Muted() config.Muted
-}
-
-// Checker waits for the application to be totally deployed.
-type Checker interface {
-	Check(context.Context, io.Writer) error
+	StatusCheck() (*bool, error)
 }
 
 // statusChecker runs status checks for pods and deployments
-type statusChecker struct {
+type Checker struct {
 	cfg             Config
 	labeller        *label.DefaultLabeller
 	deadlineSeconds int
 	muteLogs        bool
+	seenResources   resource.Group
+	singleRun       singleflight.Group
 }
 
 // NewStatusChecker returns a status checker which runs checks on deployments and pods.
-func NewStatusChecker(cfg Config, labeller *label.DefaultLabeller) Checker {
-	return statusChecker{
+func NewStatusChecker(cfg Config, labeller *label.DefaultLabeller) *Checker {
+	return &Checker{
 		muteLogs:        cfg.Muted().MuteStatusCheck(),
 		cfg:             cfg,
 		labeller:        labeller,
 		deadlineSeconds: cfg.StatusCheckDeadlineSeconds(),
+		seenResources:   make(resource.Group),
+		singleRun:       singleflight.Group{},
 	}
 }
 
-// Run runs the status checks on deployments and pods deployed in current skaffold dev iteration.
-func (s statusChecker) Check(ctx context.Context, out io.Writer) error {
+// Check runs the status checks on deployments and pods deployed in current skaffold dev iteration.
+func (s *Checker) Check(ctx context.Context, out io.Writer) error {
+	_, err, _ := s.singleRun.Do(s.labeller.GetRunID(), func() (interface{}, error) {
+		return struct{}{}, s.check(ctx, out)
+	})
```

Comment:
this signature looks very strange, returning `_, err, _` seems like we could maybe be doing this a different way?

Reply:
`singleflight.Do` is very generic where the return values `a, b, c` represent:
- a: the return value from the single run of the constituent `func`. In our case it doesn't return any value
- b: the return error from the single run of the constituent `func`. We only care about this value
- c: if there were concurrent requests at all. Again we don't care about this value.

I believe this is the right way to do this.